src/components/timeline/index.js

```diff
@@ -78,64 +106,84 @@ class Timeline extends PureComponent<Props> {
-              Categories
-            </label>
-            <label className="timelineToggleLabel">
-              <input
-                type="radio"
-                name="timelineToggle"
-                className="timelineToggleInput"
-                checked={timelineType === 'stack'}
-                onChange={this._changeToStacks}
-              />
-              Stack height
-            </label>
-          </div>
-        </form>
-        <TimelineRuler
-          zeroAt={zeroAt}
-          rangeStart={committedRange.start}
-          rangeEnd={committedRange.end}
-          width={width}
-        />
-        <OverflowEdgeIndicator
-          className="timelineOverflowEdgeIndicator"
-          panelLayoutGeneration={panelLayoutGeneration}
-        >
-          {
-            <Reorderable
-              tagName="ol"
-              className="timelineThreadList"
-              grippyClassName="timelineTrackGlobalGrippy"
-              order={globalTrackOrder}
-              orient="vertical"
-              onChangeOrder={changeGlobalTrackOrder}
-            >
-              {globalTracks.map((globalTrack, trackIndex) => (
-                <TimelineGlobalTrack
-                  key={trackIndex}
-                  trackIndex={trackIndex}
-                  trackReference={globalTrackReferences[trackIndex]}
+      <>
+        <div className="timelineSettings">
+          <form>
+            <div className="timelineSettingsToggle">
+              Graph type:{' '}
+              <label className="timelineSettingsToggleLabel">
+                <input
+                  type="radio"
+                  name="timelineSettingsToggle"
+                  className="timelineSettingsToggleInput"
+                  checked={timelineType === 'category'}
+                  onChange={this._changeToCategories}
+                />
+                Categories
+              </label>
+              <label className="timelineSettingsToggleLabel">
+                <input
+                  type="radio"
+                  name="timelineSettingsToggle"
+                  className="timelineSettingsToggleInput"
+                  checked={timelineType === 'stack'}
+                  onChange={this._changeToStacks}
                 />
-              ))}
-            </Reorderable>
-          }
-        </OverflowEdgeIndicator>
-      </TimelineSelection>
+                Stack height
+              </label>
+            </div>
+          </form>
+          <button
+            type="button"
+            onClick={this._showMenu}
+            className="timelineSettingsHiddenTracks"
+          >
+            <span className="timelineSettingsHiddenTracksNumber">
+              {hiddenTrackCount.total - hiddenTrackCount.hidden}
+            </span>
+            {' / '}
+            <span className="timelineSettingsHiddenTracksNumber">
+              {hiddenTrackCount.total}{' '}
+            </span>
+            tracks visible
+          </button>
+        </div>
```

Comment:
I think it's worth extracting this part into 2 separate components (possibly defined in this same file, I don't mind):

1. TimelineSettings: contains the radio buttons
2. HiddenTrackWidget: contains the button and controls the context menu

Reply:
I opted for the following names, to follow the name concatenation pattern:

`TimelineSettingsGraphType`
`TimelineSettingsHiddenTracks`